Does not build with boost 1.67.0

Monero does not build with boost 1.67.0.

---

[ 28%] Building CXX object contrib/epee/src/CMakeFiles/epee.dir/connection_basic.cpp.o
In file included from monero/contrib/epee/src/connection_basic.cpp:35:
In file included from monero/contrib/epee/include/net/connection_basic.hpp:63:
monero/contrib/epee/include/syncobj.h:37:10: fatal error: 'boost/thread/v2/thread.hpp' file not
      found
#include <boost/thread/v2/thread.hpp>
         ^
1 error generated.
make[3]: *** [contrib/epee/src/CMakeFiles/epee.dir/connection_basic.cpp.o] Error 1
make[2]: *** [contrib/epee/src/CMakeFiles/epee.dir/all] Error 2
make[1]: *** [all] Error 2

---

$ ls /usr/local/Cellar/boost/1.67.0/include/boost/thread/v2/
shared_mutex.hpp

--- a/contrib/epee/include/net/abstract_tcp_server2.inl
+++ b/contrib/epee/include/net/abstract_tcp_server2.inl
@@ -41,7 +41,8 @@
 #include <boost/utility/value_init.hpp>
 #include <boost/asio/deadline_timer.hpp>
 #include <boost/date_time/posix_time/posix_time.hpp> // TODO
-#include <boost/thread/v2/thread.hpp> // TODO
+#include <boost/thread/thread.hpp> // TODO
+#include <boost/thread/condition_variable.hpp> // TODO
 #include "misc_language.h"
 #include "pragma_comp_defs.h"
 

--- a/contrib/epee/include/syncobj.h
+++ b/contrib/epee/include/syncobj.h
@@ -31,10 +31,11 @@
 #define __WINH_OBJ_H__
 
 #include <boost/chrono/duration.hpp>
+#include <boost/thread/condition_variable.hpp>
 #include <boost/thread/locks.hpp>
 #include <boost/thread/mutex.hpp>
 #include <boost/thread/recursive_mutex.hpp>
-#include <boost/thread/v2/thread.hpp>
+#include <boost/thread/thread.hpp>
 
 namespace epee
 {

